### DISCUSSION TITLE
Simulations to demonstrate capabilities for cognitive neuroscience

### DISCUSSION THREAD

**Discussion by crcox** [pykale/pykale#308] [Ideas]
My aim is to generate very small datasets with known properties to observe domain adaptation under ideal circumstances. I believe this will help me develop intuitions about the technique, what kinds of information can be transferred, etc. It would be very helpful to discuss what these datasets should look like and how best to execute the simulations using PyKale.

For example, we could generate 100 source and target examples defined as 90 dimensional vectors, and specify that the relationship between the data and the category membership relies on only features `1 -- 6` in the source data and only features `3 -- 9` in the target data. There are three "unique" features in each dataset, and three that are important in both. The dimensionality of this problem is arbitrary, and I think what I would want to do is limit the data available in the target dataset (or increase the amount of noise) and see if domain adaptation improves performance and increases the likelihood of identifying features `3 -- 6`.

```
import numpy as np

D = 90
N = 100
x_source = np.random.randn(N*D).reshape(N, D)
x_target = np.random.randn(N*D).reshape(N, D)
b_star_source = np.concatenate([np.ones(6), np.zeros(D - 6)])
b_star_target = np.concatenate([np.zeros(3), np.ones(6), np.zeros(D - 9)])
y_source  = np.dot(x_source, b_star_source) + np.random.randn(N)
y_target = np.dot(x_target, b_star_target) + np.random.randn(N)
y_cat_source = np.where([y < np.median(y_source) for y in y_source], 1, 2)
y_cat_target = np.where([y < np.median(y_target) for y in y_target], 1, 2)
```

Does this setup make sense, and if so how can I hand such a simple problem to PyKale (or perhaps core functions within the library that implement domain adaptation)?

@sz144 do you have suggestions?

**Comment by shuo-zhou** [comment]
Thank you @crcox . This discussion makes us aware of the importance of toy examples for demonstrating. I will make one or two examples with linear and interpretable approaches. Hopefully it can be merged together with the example on brain imaging data.

The example suggested above is not a typical domain adaptation problem, but it is helpful for us to understand from the user end. Please let us know if you have any further suggestions. Your input and contribution is greatly appreciated.

**Reply by crcox** [reply]
Thank you @sz144 ! It is helpful to know that this is not a typical problem. Is it not typical because it is just too simple? What would be a simple problem where you could apply domain adaptation in a typical way?

**Reply by shuo-zhou** [reply]
Please see this toy dataset [example](https://domainadaptation.org/demos/salad.datasets.html#toy-datasets). We will generate similar data, but linear case. Hopefully it will explain domain adaptation better.

**Reply by shuo-zhou** [reply]
Hi @crcox, this is a [toy example](https://colab.research.google.com/github/sz144/pykale/blob/brain-example/examples/toy_domain_adaptation/tutorial.ipynb) I implemented. Please let me know if you have any comments.

**Reply by haipinglu** [reply]
> Hi Chris, this is a [toy example](https://colab.research.google.com/github/sz144/pykale/blob/brain-example/examples/toy_domain_adaptation/tutorial.ipynb) I implemented. Please let me know if you have any comments.

@sz144 We should discuss (tomorrow or next PyKale meeting) where to place such toy examples, whether as an example like others, or a dedicated ToyExample under examples to include such demos (and encourage more).

**Reply by crcox** [reply]
@sz144 Thank you! Sorry I missed that you posted these. I will work through them soon.
